### PR TITLE
widget previews

### PR DESCRIPTION
**Pull request by gregvanbrug**
## Summary

Adds the widget previews feature.

## What are the specific steps to test this change?

> 1. 

## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [ ] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [ ] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [ ] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

### CODE REVIEW

File: modules/@apostrophecms/area/index.js [hunk 1]
```diff
@@ -89,6 +100,14 @@ module.exports = {
   },
   methods(self) {
     return {
+      addExpandedMenu() {
+        self.apos.modal.add(
```
Comment (boutell) [inline modules/@apostrophecms/area/index.js:104]: Not necessary — only used when a modal is going to be instantiated via a bus event, like the manager modals.

File: modules/@apostrophecms/area/index.js [hunk 2]
```diff
@@ -140,7 +159,20 @@ module.exports = {
             in an options property.
           `);
         }
-        _.each(options.widgets, function (options, name) {
+
+        let widgets = options.widgets || {};
```
Comment (boutell) [inline modules/@apostrophecms/area/index.js:163]: Refactor and reuse this code.

**Review by boutell — CHANGES_REQUESTED** [review]
Small stuff.

**Review by boutell — APPROVED** [review]

File: modules/@apostrophecms/area/index.js [hunk 3]
```diff
@@ -43,9 +43,10 @@ module.exports = {
             throw self.apos.error('invalid');
           }
 
-          let options = field.options && field.options.widgets && field.options.widgets[type];
+          const widgets = this.getWidgets(field.options);
```
Comment (myovchev) [inline modules/@apostrophecms/area/index.js:46]: This is random failing due to a typo `this.getWidgets` should become `self.getWidgets` (it happens in several places here)

**Review by myovchev — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/area/index.js (hunk 3 above)
Comment (gregvanbrug) [reply modules/@apostrophecms/area/index.js]: 🤦‍♂️ fixed.

**Review by myovchev — APPROVED** [review]
Approving since the issues caught in the tests are fixed. The weird one reported on Slack should be reproduced first by other devs anyway.

**Review by boutell — APPROVED** [review]

**Review by boutell — APPROVED** [review]

**Pull request merged**